metagraph/src/graph/representation/canonical_dbg.cpp

```diff
@@ -62,79 +62,100 @@ void CanonicalDBG
 ::map_to_nodes_sequentially(std::string_view sequence,
                             const std::function<void(node_index)> &callback,
                             const std::function<bool()> &terminate) const {
-    std::vector<node_index> path = map_sequence_to_nodes(graph_, sequence);
-    auto first_not_found = std::find(path.begin(), path.end(), DeBruijnGraph::npos);
+    if (sequence.size() < get_k())
+        return;
+
+    std::vector<node_index> path;
+    path.reserve(sequence.size() - get_k() + 1);
+
+    // map until the first mismatch
+    bool stop = false;
+    graph_.map_to_nodes_sequentially(sequence,
```

Comment:
if you use `graph_.traverse` instead, it should be slightly faster since it'll avoid the `get_missing_kmer_skipper` call

Reply:
But get_missing_kmer_skipper is supposed to speed it up, that's why we have it.

Reply:
Ah, sorry, `get_missing_kmer_skipper` does help if there's a Bloom filter present.

What I had in mind instead was `utils::drag_and_mark_segments`. This is computed on the entire string, whereas all we need is to keep calling `fwd` and `get_edge` until we get `npos`.

Reply:
Though all of this is minor and switching to `traverse` won't really have a significant effect on mapping time.